### ISSUE TITLE
Enhancement - Update Status Reason Text

### ISSUE THREAD

**Issue by mjones-oddball** [department-of-veterans-affairs/notification-api#969]
# Description
In https://github.com/department-of-veterans-affairs/notification-api/issues/844 we added some missing status reasons, however the resulting text is a little too technical and includes variable information specific to the notification - which is more useful for logging than categorizing the type of failure. These status reasons are returned via the notification endpoint and are used for reporting/analytics in DOMO. The goal is to create reason text that business users and technical users can easily understand so they can improve errors with their notifications.

You will be able to see the status reason text by invoking the notification endpoint on an email or sms request that failed due to the applicable status reason. 

API doc: https://github.com/department-of-veterans-affairs/notification-api/blob/master/documents/openapi/openapi.yaml

Endpoint: /v2/notifications/{notification_id}

- [x] Please update the following status reasons:
- [x] QA testing
    - [x] validating each response returned via (see endpoints below)

| current text | desired text |
|---|---|
| "ERROR: MaxRetriesExceededError - 48 retries attempted but still failed to send" | Retries exceeded |
| "ERROR: InvalidProviderException - could not resolve provider" | SMS provider configuration invalid |
| "ERROR: InvalidEmailError - Email notification-id {notification_id} failed" | Email address is in invalid format |
| "ERROR: InvalidProviderException - Invalid provider for notification-id {notification_id}" | Email provider configuration invalid |

## Testing
Verify that an error occurs and the notification has the proper `status_reason` text. Can be checked with the GET notification endpoint
Send Endpoints: 
- /v2/notifications/email
- /v2/notifications/sms

GET Notification Endpoint:
- /v2/notifications/{notification-id}

---

**Comment by mjones-oddball** [comment]
Blocker Internal: pending discussion with Evan and Kyle 11/30 to understand the status reasons and provide alternative descriptions

**Comment by mjones-oddball** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/VA-Notify-620d21369d810a00146ed9c8/issues/department-of-veterans-affairs/notification-api/969?planning-poker) with Zenhub @cris-oddball @EvanParish @ianperera @jakehova @k-macmillan @kalbfled @ldraney @trevor2718

**Comment by mjones-oddball** [comment]
@mjones-oddball add notification endpoint to desc

**Comment by mjones-oddball** [comment]
@mjones-oddball new ticket to reconfigure "ERROR: NonRetryableException - permenant failure, not retrying" due to STOP (opt-out) as a preference-declined status with applicable status reason.

**Comment by mjones-oddball** [comment]
@cris-oddball idea for triggering the provider errors - you can assign a provider to a template and then mark that provider as inactive (we think that should work)

**Comment by mjones-oddball** [comment]
Updated ticket description, unblocked

**Comment by mjones-oddball** [comment]
added notification endpoint to ticket

**Comment by EvanParish** [comment]
There are status reasons that start with `ERROR: BotoClientError`. These should not be ignored. I could change them to say `Internal server error` for now. I would need to go digging deeper to see how these would actually be triggered, but for now the messages should be changed to something else. That way we aren't revealing to much about our infrastructure if they do come up, as Cris pointed out.

**Comment by EvanParish** [comment]
@cris-oddball Just a heads up that this ticket has been picked up. The code changes will be minor, just changing the error text above. The main issue will be testing and making sure all the status reasons are showing up properly.

**Comment by EvanParish** [comment]
I was able to successfully get email and sms provider configuration error messages during testing. See below. 

What's holding me up now is attempting to add the status reasons to unit tests. I'm trying to verify they can be assigned. I was able to test this successfully locally by adding the necessary feature flag (`NOTIFICATION_FAILURE_REASON_ENABLED='True'`) to my local `.env` file. When pushing to the draft PR the tests fail. I'm looking into why this feature flag isn't being set in the testing environment.

```json
{
  "status": "technical-failure", 
  "status_reason": "Email provider configuration invalid"
}
```
```json
{
  "status": "technical-failure", 
  "status_reason": "SMS provider configuration invalid"
}
```

**Comment by mjones-oddball** [comment]
@EvanParish would you be offended if I moved the query request into a new ticket? Feel like at this point it belongs separate so we can track and close out these changes.

**Comment by EvanParish** [comment]
@mjones-oddball I'm fine with that. There's just some unit tests failing that are currently holding up this ticket. We are looking into why that is.

**Comment by EvanParish** [comment]
The issue has been resolved and the PR is up for review.

**Comment by cris-oddball** [comment]
**PASSES QA**

**Retries exceeded** 
- unable to test, covered in unit tests

**Invalid email format** 
- unable to test, convered in unit tests
	
**SMS provider configuration invalid**
- change sms provider to `"active": false`, send an sms, verify notification status, then revert
- `/v2/notifications/{notification-id} `returned the following properties:
```
"status": "technical-failure",
"status_reason": "SMS provider configuration invalid",
```
    
**Email provider configuration invalid**
- change email provider to `"active": false`, send an email, verify notification status, then revert
- `/v2/notifications/{notification-id} `returned the following properties:
```
"status": "technical-failure",
"status_reason": "Email provider configuration invalid",
```
